### PR TITLE
Generic Meta macro and CSS component

### PR DESCRIPTION
**Pull request by tyom**
- Add `c-meta-container` CSS component with modifiers to configure the display style of `c-meta-item`s.
- Replace `c-meta-list` for more generic component

### CODE REVIEW

File: src/apps/investment-projects/middleware/shared.js [hunk 1]
```diff
@@ -51,7 +52,7 @@ async function getInvestmentDetails (req, res, next, id = req.params.id) {
 
     res.locals.investmentStatus = {
       id: investmentData.id,
-      meta: [
+      meta: transformFieldsObjectToMacrosObject([
```
Comment (feedmypixel) [inline src/apps/investment-projects/middleware/shared.js]: is this work missing?

Comment (tyom) [inline src/apps/investment-projects/middleware/shared.js]: Hmm.. Might have lost that one somewhere between branches.

File: assets/stylesheets/components/_meta-container.scss [hunk 2]
```diff
@@ -0,0 +1,48 @@
+@import "../settings";
+@import "../tools";
+
+.c-meta-container {
```
Comment (teneightfive) [inline assets/stylesheets/components/_meta-container.scss]: I think we should combine `.c-meta-container` and `.c-meta-item` seeing as the container is always included when adding meta items now.

The modifiers in this file aren't modifying this block they are modifying a different block which doesn't feel right to me.

File: src/apps/investment-projects/middleware/shared.js (hunk 1 above)
Comment (tyom) [inline src/apps/investment-projects/middleware/shared.js]: Just realised that the function was added in #447 (`simplify-macro-config` branch) which is what this branch depends on.

File: assets/stylesheets/components/_meta-list.scss [hunk 3]
```diff
@@ -1,11 +1,75 @@
...
     margin-left: $default-spacing-unit;
   }
 
-  .c-meta-item__label {
+  .c-meta-list__item-label {
     display: block;
   }
 }
+
+.c-meta-list--alternate {
```
Comment (teneightfive) [inline assets/stylesheets/components/_meta-list.scss]: Wonder if there's a better name than `alternate`. Alternate to what? Maybe `--split` or something to say that they're floated right and left.

File: config/nunjucks/filters.js [hunk 4]
```diff
@@ -139,6 +139,15 @@ const filters = {
     }
     return result
   },
+
+  classModifier (className, modifier) {
```
Comment (teneightfive) [inline config/nunjucks/filters.js]: 👏 but what about `applyClassModifiers`? Bit more specific as to what its doing as its not actually modifying the class you give it.

**Pull request merged**